File: algorithms/strings/urlify/URLify.java
public class Urlify {

    private static void urlify(char[] arr, int trueLength) {
        int index = arr.length;

        for (int i = trueLength - 1; i >= 0; i--) {
            if (arr[i] == ' ') {
                arr[index - 1] = '0';
                arr[index - 2] = '2';
                arr[index - 3] = '%';
                index -= 3;
            } else {
                arr[--index] = arr[i];
            }
        }
    }

    public static void main(String[] args) {
        String str = "Mr John Smith    ";
        char[] arr = str.toCharArray();

        urlify(arr, 13);

        System.out.println("Encoded string : " + new String(arr));
    }

}

File: algorithms/strings/urlify/urlify.cpp
#include <iostream>
using namespace std;

void urlify(char *arr, int true_length)
{
    int actual_idx = sizeof(arr) / sizeof(arr[0]);

    for (int i = true_length - 1; i >= 0; i--)
    {
        if (arr[i] == ' ')
        {
            arr[actual_idx - 1] = '0';
            arr[actual_idx - 2] = '2';
            arr[actual_idx - 3] = '%';
            actual_idx -= 3;
        }
        else
        {
            arr[--actual_idx] = arr[i];
        }
    }
}

int main()
{
    char str[] = "Mr John Smith    ";
    urlify(str, 13);

    cout << "Encoded string : " << str << "\n";

    return 0;
}